### PR TITLE
Replaced PDO Shadow Bias With New Shadow Factor

### PR DESCRIPTION
**Pull request by santorac**
There was a special hard-coded shadow depth bias applied to all parallax PDO. It was ineffective because it was too large and didn't take the geometric slope into account. This was causing incorrect results in AtomSampleViewer's parallax screenshot tests, where self-shadows from a spotlight didn't appear at all on the parallax surface.

According to what history I can find, this bias was added because PDO surfaces were causing shadow acne, and apparently standard shadow biasing didn't work on PDO surfaces, at least at that time (see https://github.com/o3de/o3de/pull/1558 and the code comments in Shadow.azsli). Well that's not true anymore, both standard shadow bias and normal shadow bias are working on parallax PDO surfaces. However, since parallax surfaces are much more prone to artifacts than normal surface types, we shouldn't expect users to increase bias for all their lights to accommodate these surfaces.

I added a new PDO specific "Shadow Factor" setting to all relevant material types to better address acne or other artifacts. This factor works by squashing the displaced surface further down during the shadowmap pass.

https://user-images.githubusercontent.com/55155825/166085856-92c3a0f5-271e-4a20-997f-39e591230ff4.mp4

Also made some improvements to support debugging projected shadow maps. 
- Added min/max sliders to the ImGuiPassTree to support trimming the color output for increased contrast of fine details. 
- Stopped the ProjectedShadowFeatureProcessor from rebuilding the shadowmap passes every time the light position was updated. When using the pass tree to preview the projected shadow map output, this was recreating the preview attachment every frame and quickly overflowed the AttachmentImage budget.

Testing:
- I set up a test scene with a cloth material using parallax PDO, and a spotlight, per the old bug report record associated with this issue. When I removed the PdoDepthBias, the acne appeared, and then when I adjuste the spotlight's bias and/or normal bias the acne was successfully resolved.
- Ran the AtomSampleViewer parallax test script and visiuall confirmed that the shadows have returned, matching closer to the original expected screenshots. I will update the ASV screenshots with new versions using my latest results.
- AtomSampleViewer automation passed (with update parallax screenshots) on dx12 and vulkan, except for one known issue.\
- Added a spotlight with shadow in the level editor, and updated each of the settings on the component while verifying a corresponding change was seen in the scene, to make sure the changes to the projected shadowmap feature processor didn't break something.

**Original state before any changes:**
PdoDepthBias001

**Acne can be seed after removing the PdoDepthBias**
PdoDepthBias0

**The acne is clear again after adjusting the new Shadow Factor**
image

**Note the acne is can be addressed by adjusting the light's bias values**
PdoWithBasicAndNormalShadowBias

**Self-shadows can be seen again in the parallax test**
screenshot_7

### CODE REVIEW

**Review by jeremyong-az — APPROVED** [review]
Nice find!

File: Gems/Atom/RPI/Assets/Shaders/ImagePreview.azsl [hunk 1]
```diff
@@ -89,5 +91,8 @@ PSOutput MainPS(VSOutput psInput)
             break;
         }
     }  
+
+    OUT.m_color.xyz = (OUT.m_color.xyz - float3(PassSrg::m_colorRangeMinMax[0],PassSrg::m_colorRangeMinMax[0],PassSrg::m_colorRangeMinMax[0])) / (PassSrg::m_colorRangeMinMax[1] - PassSrg::m_colorRangeMinMax[0]);
```
Comment (antonmic) [inline Gems/Atom/RPI/Assets/Shaders/ImagePreview.azsl:95]: Spaces after the commas and a comment would make this a lot more readable 🙂

Comment (antonmic) [reply Gems/Atom/RPI/Assets/Shaders/ImagePreview.azsl:95]: btw, you can also do
PassSrg::m_colorRangeMinMax[0].xxx
If you want to create a float3 out of a single float:
`OUT.m_color.xyz = (OUT.m_color.xyz - PassSrg::m_colorRangeMinMax[0].xxx) / (PassSrg::m_colorRangeMinMax[1] - PassSrg::m_colorRangeMinMax[0]);`

**Review by antonmic — APPROVED** [review]
Awesome work! Just the one line in ImagePreview.azsl could be improved for readability, otherwise ship it!

File: Gems/Atom/RPI/Assets/Shaders/ImagePreview.azsl (hunk 1 above)
Comment (santorac) [reply Gems/Atom/RPI/Assets/Shaders/ImagePreview.azsl:95]: Right, thanks. I was moving fast and forgot to clean this up. The xxx thing escaped me at the  moment too, thanks for the reminder.

Comment (jeremyong-az) [reply Gems/Atom/RPI/Assets/Shaders/ImagePreview.azsl:95]: You can subtract scalars from vectors, shouldn't need the `.xxx` at all I believe

File: Gems/Atom/Feature/Common/Assets/Materials/Types/MaterialInputs/ParallaxPropertyGroup.json [hunk 2]
```diff
@@ -93,6 +93,19 @@
                 "name": "o_parallax_enablePixelDepthOffset"
             }
         },
+        {
+            "name": "shadowFactor",
+            "displayName": "Shadow Factor",
+            "description": "Can be used to reduce the impact of shelf-shadowing on surfaces with pixel depth offset (PDO), to reduce shadow acne or other artifacts.",
```
Comment (jeremyong-az) [inline Gems/Atom/Feature/Common/Assets/Materials/Types/MaterialInputs/ParallaxPropertyGroup.json:99]: "self-shadowing"

File: Gems/Atom/Feature/Common/Assets/Materials/Types/StandardMultilayerPBR.materialtype [hunk 3]
```diff
@@ -425,6 +425,19 @@
                         "name": "o_parallax_enablePixelDepthOffset"
                     }
                 },
+                {
+                    "name": "shadowFactor",
+                    "displayName": "Shadow Factor",
+                    "description": "Can be used to reduce the impact of shelf-shadowing on surfaces with pixel depth offset (PDO), to reduce shadow acne or other artifacts.",
```
Comment (jeremyong-az) [inline Gems/Atom/Feature/Common/Assets/Materials/Types/StandardMultilayerPBR.materialtype:431]: "self-shadowing"

File: Gems/Atom/Feature/Common/Assets/Materials/Types/MaterialInputs/ParallaxPropertyGroup.json (hunk 2 above)
Comment (jeremyong-az) [reply Gems/Atom/Feature/Common/Assets/Materials/Types/MaterialInputs/ParallaxPropertyGroup.json:99]: I'd recommend describing how the range behaves (the significance of 0 and 1 here are somewhat nebulous, although it's clear from the documentation on `AdjustParallaxShadowCaster` itself).

**Review by HogJonny-AMZN — APPROVED** [review]

**Pull request merged**